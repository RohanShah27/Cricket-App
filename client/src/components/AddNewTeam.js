import React, { Component } from "react";
import { addTeam } from "../actions/User";

import { connect } from "react-redux";
import "../styles/Adminaddnewteam.css";
//initial state for the values used in this component -yash
const initialState = {
    team_name: "",
    team_name_Error: ""
}
export class AddNewTeam extends Component {
    constructor(props) {
        super(props);
        //to make function onaddAdminClick execute when clicked  -yash
        this.onaddAdminClick = this.onaddAdminClick.bind(this);
        this.state = {
            isDisabled: false
        }
    }
    state = {
        //parameter in table teams -yash
        initialState
    };

    OnChange = event => {
        //checkbox to check if all error conditions are surpassed -yash
        const isCheckbox = event.target.type === "checkbox";
        //state change when the user inputs in the inputbox -yash
        this.setState({ [event.target.name]: isCheckbox ? event.target.checked : event.target.value });
    };
    componentDidMount() {
        //check if token is present -yash
        if (!localStorage.getItem("token")) {
            this.props.history.push("/");
        }
    }
    //errors generated from the client side -yash
    validate = () => {
        let team_name_Error = "";

        if (!this.state.team_name) {
            team_name_Error = "Team Name Cannot Be Empty";
        }

        if (team_name_Error) {
            this.setState({ team_name_Error: team_name_Error });

            return false;
        }
        return true;
    }

    onaddAdminClick(e) {
        //prevent values showing on the searchbar -yash
        e.preventDefault();
        const isValid = this.validate();
        if (isValid) {
            let team = {
                team_name: this.state.team_name
            };
            //function on actions to add the team_name to the team database -yash
            this.props.addTeam(team)
            this.setState({
                team_name: ""
            });
        }
    }

    render() {
        console.log(this.props)
        return (
            //start of div -yash
            <div>
                {/* start of form addnewteamform -yash*/}
                <form id="addnewteamform">
                    {/* start of fieldset -yash*/}
                    <fieldset>
                        {/* start and end of header -yash */}
                        <h1 className="teamadminheader">Add New Team</h1>
                        {/* start of input team-name -yash */}
                        <input
                            className="teamadminnewinput"
                            type="text"
                            name="team_name"
                            placeholder="Enter New Team Name"
                            onChange={this.OnChange}
                            value={this.state.team_name}
                        />
                        {/* end of input team-name -yash */}
                        {/* error generation if there is an error -yash */}
                        <div className="teamnameerror" style={{ fontSize: 15, color: "red" }}>
                            {/* end of error generation -yash */}
                            {/* display error from node -yash */}
                            {this.props.error ? <><p>{this.props.error}</p></> : <p>{this.state.team_name_Error}</p>}
                            {/* á¹£tart of button add team -yash */}
                        </div>
                        <button
                            onChange={this.onChange}
                            onClick={this.onaddAdminClick}
                            className="teamadmin-new-button"
                        >
                            Add Team
            </button>
                        {/* end of button add team -yash */}
                    </fieldset>
                    {/* end of fieldset -yash */}
                </form>
                {/* end of form -yash */}
            </div>
            // end of div -yash
        );
    }
}

const mapStateToProps = state => ({
    users: state.userReducer.users,
    error: state.userReducer.error
});

export default connect(
    mapStateToProps,
    //function add team in actions  -yash
    { addTeam }
)(AddNewTeam);
