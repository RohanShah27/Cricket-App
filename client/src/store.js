import { createStore, applyMiddleware, compose } from "redux";

import thunk from "redux-thunk";
import rootReducer from "./reducers";

const initialState = {};
//think will give all redducer actions to the middleware object
const middleware = [thunk];
// will give access to the extension
//   //We do this to check if chrome has the redux extension or not if not then it installs the devtoolextension in chrome
const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
      })
    : compose;
const enhancer = composeEnhancers(applyMiddleware(...middleware));
const store = createStore(rootReducer, initialState, enhancer);
export default store;
