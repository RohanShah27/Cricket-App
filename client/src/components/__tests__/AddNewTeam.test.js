import React from "react";
import { mount } from "enzyme";
import AddNewTeam from "../AddNewTeam";
import { Provider } from "react-redux";
import store from "../../store";

const AddnewTeam = jest.fn();
const wrapper = mount(
  <Provider store={store}>
    <AddNewTeam addnewteam={AddnewTeam} history={[]} />
  </Provider>
);

describe("Test AddNewTeam Component", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have exactly one input fields", () => {
    expect(wrapper.find("input").length).toBe(1);
  });

  it("should have exactly one button", () => {
    expect(wrapper.find("button").length).toBe(1);
  });

  it("should have exactly add user text on Button", () => {
    expect(wrapper.find("button").text()).toEqual("Add Team");
  });

  it("should have Team Name placeholder on team name input field", () => {
    expect(
      wrapper
        .find("input")
        .at(0)
        .props().placeholder
    ).toBe("Enter New Team Name");
  });
});
