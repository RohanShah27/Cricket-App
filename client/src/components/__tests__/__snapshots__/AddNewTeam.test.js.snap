// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddNewTeam Component should render the component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(AddNewTeam)
    addnewteam={[MockFunction]}
    history={
      Array [
        "/",
      ]
    }
  >
    <AddNewTeam
      addTeam={[Function]}
      addnewteam={[MockFunction]}
      error=""
      history={
        Array [
          "/",
        ]
      }
      users={Array []}
    >
      <div>
        <form
          id="addnewteamform"
        >
          <fieldset>
            <h1
              className="teamadminheader"
            >
              Add New Team
            </h1>
            <input
              className="teamadminnewinput"
              name="team_name"
              onChange={[Function]}
              placeholder="Enter New Team Name"
              type="text"
            />
            <div
              className="teamnameerror"
              style={
                Object {
                  "color": "red",
                  "fontSize": 15,
                }
              }
            >
              <p />
            </div>
            <button
              className="teamadmin-new-button"
              onClick={[Function]}
            >
              Add Team
            </button>
          </fieldset>
        </form>
      </div>
    </AddNewTeam>
  </Connect(AddNewTeam)>
</Provider>
`;
